tools_lib/src/stack_collapse.rs

```diff
@@ -56,116 +56,170 @@ mod test {
 
     #[test]
     fn basic_test() {
+        //                                         <--e1-->
+        //                 <--e1-->        <----------e2---------->
+        //              T2 1       2       3       4       5       6
+        // sample interval |   |   |   |   |   |   |   |   |   |   |
+        // stacks count:
+        // rustc                       1   2
+        // rustc;e1            1   2
+        // rustc;e2                            1   2           3   4
+        // rustc;e2;e3                                 1   2
```

Comment:
think this shall be e1 want to use the same event to show that it is on the stack here
```suggestion
        // rustc;e2;e1                                 1   2
```

Reply:
:+1: